### PR TITLE
[ENG-1157] [ENG-1115] List item on sale Solana

### PR DESCRIPTION
**Pull request by viet10101988**
- See README SOLANA

### CODE REVIEW

**Comment by viet10101988** [comment]
@Superjo149 you can review again

File: src/nft/SOLNFT.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        ];
+      }
+      winnerLimit = new WinnerLimit({
+        type: WinnerLimitType.Capped,
+        usize: new BN(attributes.editions || 1),
+      });
+    
+      const auctionSettings: IPartialCreateAuctionArgs = {
+        winners: winnerLimit,
+        endAuctionAt: isInstantSale
```
Comment (Superjo149) [inline src/nft/SOLNFT.ts:57]: For simplicity, shall we just assume all the logic here is `isInstantSale` and remove the conditionals? If we need any other special features, we can add it later

File: src/nft/SOLNFT.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                  : 60), // 1 minute in seconds
+            ),
+        priceFloor: new PriceFloor({
+          // type: attributes.priceFloor
+          //   ? PriceFloorType.Minimum
+          //   : PriceFloorType.None,
+          type: PriceFloorType.Minimum,
+          minPrice: new BN((attributes.priceFloor || 0) * LAMPORTS_PER_SOL),
+        }),
+        tokenMint: QUOTE_MINT.toBase58(),
```
Comment (Superjo149) [inline src/nft/SOLNFT.ts:84]: THis should be the publickKey of the token, in this case we'll probably put it in the tokenId field. So this should be `this.item.tokenId`.

File: src/nft/SOLNFT.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+          item.winningConfigType =
+            item.metadata.info.updateAuthority ===
+            (this.refinable.provider?.publicKey || SystemProgram.programId).toBase58()
+              ? WinningConfigType.FullRightsTransfer
+              : WinningConfigType.TokenOnlyTransfer;
+        }
+        item.amountRanges = [
+          new AmountRange({
+            amount: new BN(1),
+            length: new BN(attributes.editions || 1),
```
Comment (Superjo149) [inline src/nft/SOLNFT.ts:46]: Can you add a TODO for supporting editions?

File: src/nft/SOLNFT.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        name: null,
+      };
+    
+      const tieredAttributes = {
+        items: [],
+        tiers: [],
+      }
+    
+      await createAuctionManager(this.refinable.connection, this.refinable.provider, whitelistedCreatorsByCreator, auctionSettings, 
+        attributes.category === AuctionCategory.Open
```
Comment (Superjo149) [inline src/nft/SOLNFT.ts:101]: Same here, there is too much conditional logic which we do not need for instant sale

**Review by Superjo149 — CHANGES_REQUESTED** [review]
Looks much better, still don't really like having so much of the oyster stuff in our lib, but this can do for now. 

Just have a couple of small comments of the SOLNFT  code

**Comment by Superjo149** [comment]
Closing in favour of the upcoming PR

**Pull request closed without merging**